pandas/core/groupby/generic.py

```diff
@@ -1546,14 +1546,14 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             if self._grouper.nkeys > 1:
                 # test_groupby_as_index_series_scalar gets here with 'not self.as_index'
                 return self._python_agg_general(func, *args, **kwargs)
-            elif args or kwargs:
+            elif (args or kwargs) and (len(self._obj_with_exclusions.columns) == 1):
                 # test_pass_args_kwargs gets here (with and without as_index)
                 # can't return early
                 result = self._aggregate_frame(func, *args, **kwargs)
 
             else:
                 # try to treat as if we are passing a list
-                gba = GroupByApply(self, [func], args=(), kwargs={})
+                gba = GroupByApply(self, [func], args=args, kwargs=kwargs)
```

Comment:
The removal of args and kwargs at this point seems to have been a design decision made in 2014 in https://github.com/pandas-dev/pandas/pull/6718 to address a bug when using UDF

There have been considerable changes since and `agg` generally seems to pass on kwargs well now. When updating just this line, all tests passed so I assume this change would be okay.

Reply:
Agreed that this is okay - but not just because all tests pass. It is not good behavior to silently ignore args/kwargs, and it seems to me users are reasonable to expect that these would be passed through.